pkg/urbit/jets/b/murn.c

```diff
@@ -3,54 +3,50 @@
 */
 #include "all.h"
 
-  u3_noun
-  _murn_in(u3j_site* sit_u, u3_noun a)
-  {
-    if ( 0 == a ) {
-      return a;
-    }
-    else if ( c3n == u3du(a) ) {
-      return u3m_bail(c3__exit);
-    }
-    else {
-      u3_noun one = u3j_gate_slam(sit_u, u3k(u3h(a)));
-      u3_noun two = _murn_in(sit_u, u3t(a));
-      u3_noun nex;
-
-      switch ( u3ud(one) ) {
-        case c3y:  u3z(one);
-                   return two;
-        case c3n:  nex = u3nc(u3k(u3t(one)), two);
-                   u3z(one);
-                   return nex;
-        default:   u3z(one);
-                   u3z(two);
-                   return u3_none;
-      }
-    }
-  }
+u3_noun
+u3qb_murn(u3_noun a, u3_noun b)
+{
+  u3_noun pro = u3_nul;
 
-/* functions
-*/
-  u3_noun
-  u3qb_murn(u3_noun a, u3_noun b)
-  {
-    u3_noun pro;
+  if ( u3_nul == a ) {
+    return u3_nul;
+  }
+  else {
+    u3_noun    pro;
+    u3_noun*   lit = &pro;
     u3j_site sit_u;
+
     u3j_gate_prep(&sit_u, u3k(b));
-    pro = _murn_in(&sit_u, a);
+    {
+      u3_noun* hed;
+      u3_noun* tel;
+      u3_noun  res, i, t = a;
+
+      do {
+        u3x_cell(t, &i, &t);
+
+        res = u3j_gate_slam(&sit_u, u3k(i));
+
+        if ( u3_nul != res ) {
+          *lit = u3i_defcons(&hed, &tel);
+          *hed = u3t(res);
```

Comment:
Per the original `_murn_in`, you want `u3k(hed)` and `u3z(res)`, I don't think `u3t` does anything to reference counts.

Reply:
Good call, this leaks the outer cell.